angularjs/angular-mocks.d.ts

```diff
@@ -97,137 +97,208 @@ declare module angular {
     // see https://docs.angularjs.org/api/ngMock/service/$httpBackend
     ///////////////////////////////////////////////////////////////////////////
     interface IHttpBackendService {
+        /**
+          * Flushes all pending requests using the trained responses.
+          * @param count Number of responses to flush (in the order they arrived). If undefined, all pending requests will be flushed. 
+          */
         flush(count?: number): void;
+        
+        /**
+          * Resets all request expectations, but preserves all backend definitions.
+          */
         resetExpectations(): void;
+        
+        /**
+          * Verifies that all of the requests defined via the expect api were made. If any of the requests were not made, verifyNoOutstandingExpectation throws an exception.
+          */
         verifyNoOutstandingExpectation(): void;
+        
+        /**
+          * Verifies that there are no outstanding requests that need to be flushed.
+          */
         verifyNoOutstandingRequest(): void;
 
-        expect(method: string, url: string, data?: string, headers?: Object): mock.IRequestHandler;

```

Comment:
Had I used overloads for everything, adding the missing (url) => bool parameter option to all expect and when flavors we would have 140ish overloads, with documentation needed for each. I elected to use *unions*, which match what angular documented. This reduced the signature count to 16. I added documentation to all of those signatures. I don't know that I want to document 140 signatures.